test/unit/record/model_definition_test.rb

```diff
@@ -159,7 +159,97 @@ def setup
       assert_equal(value, @record[:name])
       assert_equal(value, @record.name)
     end
-    
+
+    should "use declared value_if_nil when value is nil or key is omitted" do
+      unique_truthy_value_if_nil = 'NONSENSE'
+
+      @record.class_eval {
+        define_simple_attribute(:string_field, :string, {}, unique_truthy_value_if_nil)
+      }
+
+      assert_equal(nil, @record.attributes[:string_field])
+      assert_equal(unique_truthy_value_if_nil, @record.string_field)
+      @record.attributes[:string_field] = nil
+      assert_equal(nil, @record.attributes[:string_field])
+      assert_equal(unique_truthy_value_if_nil, @record.string_field)
```

Comment:
I thought to test both cases of attribute missing as well as value equal nil, not sure if you want to keep that in the test suite. Also, you may want to consolidate the class_eval's